### PR TITLE
launch debug experience for csharp and node

### PR DESCRIPTION
**Pull request by adashen**

### CODE REVIEW

**Comment by formulahendry** [comment]
Fix tslint error
```
ERROR: (semicolon) src/common/constants.ts[96, 60]: Missing semicolon
ERROR: (ordered-imports) src/edge/edgeManager.ts[9, 1]: Import sources within a group must be alphabetized.
ERROR: (quotemark) src/edge/edgeManager.ts[187, 53]: ' should be "
ERROR: (prefer-const) src/edge/edgeManager.ts[190, 13]: Identifier 'outputFile' is never reassigned; use 'const' instead of 'let'.
ERROR: (no-trailing-whitespace) src/edge/edgeManager.ts[195, 127]: trailing whitespace
ERROR: (arrow-parens) src/edge/edgeManager.ts[278, 80]: Parentheses are required around the parameters of an arrow function definition
```

File: package.json [hunk 1]
```diff
@@ -117,6 +129,16 @@
           "type": "string",
           "default": "",
           "description": "(Windows Only) Replace the Windows style drive letter in the command with a Unix style root when using a custom shell as the terminal, like Bash on Windows, Git Bash or Cgywin. Example: For Bash on Windows, setting this to '/mnt/' will replace 'C:\\foo\\bar' with '/mnt/c/foo/bar'"
+        },
+        "azure-iot-edge.EdgeHubConnectionString": {
+          "type": "string",
+          "default": "<insert your Module Connection String>",
```
Comment (formulahendry) [inline package.json]: Would user manually set it? If not, we could just use `"default": ""`

File: package.json [hunk 2]
```diff
@@ -117,6 +129,16 @@
...
+          "description": "Module Connection String"
+        },
+        "azure-iot-edge.EdgeModuleCACertificateFile": {
+          "type": "string",
+          "default": "<insert your Edge Module CA path>",
```
Comment (formulahendry) [inline package.json]: Same here

File: assets/solution/launch_csharp.json [hunk 3]
```diff
@@ -23,6 +23,21 @@
         "%APP_FOLDER%": "${workspaceFolder}/modules/%MODULE_FOLDER%"
       },
       "justMyCode": true
+    },
+    {
+      "name": "%MODULE% Local Debug (.NET Core)",
```
[the remark this answers is not in the document]
Comment (adashen) [reply assets/solution/launch_csharp.json]: The naming here is from the pre-configured .net debug configuration in vscode. (.NET Core Launch (web)). So I prefer use the same naming pattern here

[the remark this answers is not in the document]
Comment (LazarusX) [reply assets/solution/launch_csharp.json]: I see.

File: src/edge/edgeManager.ts [hunk 4]
```diff
@@ -196,6 +201,26 @@ export class EdgeManager {
         } catch (err) {}
     }
 
+    public async startEdgeHubSingleModule(outputChannel: vscode.OutputChannel): Promise<void> {
+        const inputs = await this.inputInputNames();
+        await Executor.executeCMD(outputChannel, "iotedgehubdev", {shell: true}, `start -i "${inputs}"`);
```
Comment (LazarusX) [inline src/edge/edgeManager.ts]: `iotedgedev start -i` requires sudo on *nix systems.

File: src/edge/edgeManager.ts [hunk 5]
```diff
@@ -196,6 +201,26 @@ export class EdgeManager {
...
+    }
+
+    public async setModuleCred(outputChannel: vscode.OutputChannel): Promise<void> {
+        let storagePath = this.context.storagePath;
+        if (!storagePath) {
+            storagePath = path.resolve(os.tmpdir(), "vscodeedge");
+        }
+        await fse.ensureDir(storagePath);
+        const outputFile = path.join(storagePath, "module.env");
+        await Executor.executeCMD(outputChannel, "iotedgehubdev", {shell: true}, `modulecred -o ${outputFile}`);
```
Comment (LazarusX) [inline src/edge/edgeManager.ts]: `iotedgedev start -i` requires sudo on *nix systems.

**Pull request merged**